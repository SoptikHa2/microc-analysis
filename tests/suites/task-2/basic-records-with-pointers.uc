 f() {
   var n,a,b;
   b = null;
   n = 42;

   while (n>0) {
     n = n-1;
     a = alloc {h: n, t: b};
     b = a;
   }

   return 0;
 }
