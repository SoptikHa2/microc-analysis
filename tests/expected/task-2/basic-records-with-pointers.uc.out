---
test: type suites/task-2/basic-records-with-pointers.uc
---
⟦f(){...}:[1:2]⟧ = () -> int
⟦n[2:8]⟧ = int
⟦a[2:10]⟧ = ↑µt2.{h:int,t:↑t2}
⟦b[2:12]⟧ = ↑µt2.{h:int,t:↑t2}

exit: 0
